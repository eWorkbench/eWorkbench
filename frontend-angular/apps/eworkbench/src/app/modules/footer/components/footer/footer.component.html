<!--
  Copyright (C) 2016-2020 TU Muenchen and contributors of ANEXIA Internetdienstleistungs GmbH
  SPDX-License-Identifier: AGPL-3.0-or-later
-->

<ng-container *transloco="let t">
  <footer class="navbar navbar-expand-lg py-0">
    <span class="navbar-brand">{{ t('footer.brand') }}</span>
    <ul class="navbar-nav ml-auto h-100">
      <li class="h-100 d-flex align-items-center" *ngIf="currentUser">
        <a class="h-100 w-100" [routerLink]="['/contact']">
          <button class="btn ewb-button-regular font-weight-normal h-100 w-100 text-left">
            {{ t('footer.menu.contact') }}
          </button>
        </a>
      </li>
      <li class="h-100 d-flex align-items-center">
        <a class="h-100 w-100" [routerLink]="['/faq']">
          <button class="btn ewb-button-regular font-weight-normal h-100 w-100 text-left">
            {{ t('footer.menu.faq') }}
          </button>
        </a>
      </li>
      <li class="h-100 d-flex align-items-center">
        <a class="h-100 w-100" [routerLink]="['/privacy-policy']">
          <button class="btn ewb-button-regular font-weight-normal h-100 w-100 text-left">
            {{ t('footer.menu.privacyPolicy') }}
          </button>
        </a>
      </li>
      <!-- <li class="h-100 d-flex align-items-center">
        <a class="h-100 w-100" [routerLink]="['/accessibility']">
          <button class="btn ewb-button-regular font-weight-normal h-100 w-100 text-left">{{ t('footer.menu.accessibility') }}</button>
        </a>
      </li> -->
      <li class="h-100 d-flex align-items-center">
        <a class="h-100 w-100" [routerLink]="['/imprint']">
          <button class="btn ewb-button-regular font-weight-normal h-100 w-100 text-left">
            {{ t('footer.menu.imprint') }}
          </button>
        </a>
      </li>
      <li class="h-100 d-flex align-items-center">
        <a class="h-100 w-100" [routerLink]="['/licenses']">
          <button class="btn ewb-button-regular font-weight-normal h-100 w-100 text-left">
            {{ t('footer.menu.licenses') }}
          </button>
        </a>
      </li>
      <li class="h-100 d-flex align-items-center">
        <a class="h-100 w-100" href="https://github.com/eworkbench/" target="_blank">
          <button class="btn ewb-button-regular font-weight-normal h-100 w-100 text-left">
            {{ t('footer.menu.github') }}
          </button>
        </a>
      </li>
    </ul>
  </footer>
</ng-container>
