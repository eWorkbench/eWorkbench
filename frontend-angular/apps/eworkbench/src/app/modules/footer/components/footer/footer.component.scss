/**
 * Copyright (C) 2016-2020 TU Muenchen and contributors of ANEXIA Internetdienstleistungs GmbH
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

@import 'colors';
@import 'variables';

$footer-padding-wide: 68px;
$footer-padding-narrow: 20px;
$footer-text-color: #595959;

:host {
  display: block;
  z-index: 1;

  footer {
    background-color: $secondary-color;
    color: $footer-text-color;
    border-top: 1px solid $border-color;

    @media (min-width: 1062px) {
      height: 50px;
    }

    .navbar-brand {
      font-size: $font-size-sm;
      padding-top: 8px;
      border-bottom: 3px solid transparent;
      white-space: pre-wrap;

      @media (min-width: 1062px) {
        word-wrap: unset;
      }
    }

    &.navbar {
      padding-left: $footer-padding-narrow;
      padding-right: $footer-padding-narrow;

      @media (min-width: 1062px) {
        padding-left: $footer-padding-wide;
        padding-right: $footer-padding-wide;
      }
    }

    ul {
      width: 100%;

      @media (min-width: 1062px) {
        width: unset;
      }

      li {
        padding: 4px 0;

        a {
          .ewb-button-regular {
            color: $footer-text-color;
            padding: 0;

            @media (min-width: 1062px) {
              padding: 0 10px;
            }

            &:focus {
              box-shadow: none;
            }
          }
        }

        @media (min-width: 1062px) {
          border-bottom: 3px solid transparent;
        }

        &:hover {
          @media (min-width: 1062px) {
            border-color: $primary-color;
          }
        }
      }
    }
  }
}
