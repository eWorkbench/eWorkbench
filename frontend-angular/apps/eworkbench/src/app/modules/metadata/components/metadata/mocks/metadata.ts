/**
 * Copyright (C) 2016-2020 TU Muenchen and contributors of ANEXIA Internetdienstleistungs GmbH
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import type { Metadata, MetadataField } from '@eworkbench/types';

interface ParametersData {
  [id: string]: MetadataField;
}

export const mockMetadata: Metadata = {
  content_type: 59,
  content_type_model: 'metadata.metadata',
  id: '8ce37733-6abd-4f08-a5e9-1fb92d554af4',
  display: '5ea5649a-6930-4554-b972-8235da883fbb',
  field: 'e22cf025-c10f-4097-aef7-d9ea9a802562',
  field_info: {
    base_type: 'selection',
    content_type: 60,
    content_type_model: 'metadata.metadatafield',
    created_at: '2019-05-28T13:55:27.801603+02:00',
    created_by: {
      pk: 150,
      username: 'testuser',
      email: 'testuser@domain.com',
      is_active: true,
      is_staff: true,
      last_login: '2020-07-02T15:06:38.181948+02:00',
      userprofile: {
        first_name: 'Test',
        last_name: 'User',
        anonymized: false,
        academic_title: 'Test',
        additional_information: 'Something about me',
        country: 'Österreich',
        email_others: [],
        org_zug_mitarbeiter: [],
        org_zug_mitarbeiter_lang: ['Aff1edit', 'Aff1', '2', '3'],
        org_zug_student: [],
        org_zug_student_lang: ['AffS1', 'AffS2edit', '1', '3'],
        phone: '+43 123 45678963421',
        salutation: '',
        title_salutation: '',
        title_pre: '',
        title_post: '',
        type: 'u',
        avatar: '',
        website: '',
      },
    },
    description: 'test',
    display: 'Check-text',
    last_modified_at: '2019-06-18T09:57:24.704395+02:00',
    last_modified_by: {
      pk: 150,
      username: 'testuser',
      email: 'testuser@domain.com',
      is_active: true,
      is_staff: true,
      last_login: '2020-07-02T15:06:38.181948+02:00',
      userprofile: {
        first_name: 'Test',
        last_name: 'User',
        anonymized: false,
        academic_title: 'Test',
        additional_information: 'Something about me',
        country: 'Österreich',
        email_others: [],
        org_zug_mitarbeiter: [],
        org_zug_mitarbeiter_lang: ['Aff1edit', 'Aff1', '2', '3'],
        org_zug_student: [],
        org_zug_student_lang: ['AffS1', 'AffS2edit', '1', '3'],
        phone: '+43 123 45678963421',
        salutation: '',
        title_salutation: '',
        title_pre: '',
        title_post: '',
        type: 'u',
        avatar: '',
        website: '',
      },
    },
    name: 'Check-text',
    pk: 'e22cf025-c10f-4097-aef7-d9ea9a802562',
    type_settings: {
      final: true,
      answers: ['Check me'],
      multiple_select: true,
    },
  },
  ordering: 1,
  pk: '5ea5649a-6930-4554-b972-8235da883fbb',
  values: {
    answers: [
      {
        answer: 'Check me',
        selected: true,
      },
    ],
  },
};

export const mockMetadata2: Metadata = {
  content_type: 59,
  content_type_model: 'metadata.metadata',
  id: '7e1f468a-a221-407d-9eb8-bddcd707656f',
  display: 'df916528-90cf-4e00-9353-f44123c74711',
  field: '9d33359a-afd7-4149-bbe0-95f819d6bebd',
  field_info: {
    base_type: 'selection',
    content_type: 60,
    content_type_model: 'metadata.metadatafield',
    created_at: '2019-05-28T13:55:27.801603+02:00',
    created_by: {
      pk: 150,
      username: 'testuser',
      email: 'testuser@domain.com',
      is_active: true,
      is_staff: true,
      last_login: '2020-07-02T15:06:38.181948+02:00',
      userprofile: {
        first_name: 'Test',
        last_name: 'User',
        anonymized: false,
        academic_title: 'Test',
        additional_information: 'Something about me',
        country: 'Österreich',
        email_others: [],
        org_zug_mitarbeiter: [],
        org_zug_mitarbeiter_lang: ['Aff1edit', 'Aff1', '2', '3'],
        org_zug_student: [],
        org_zug_student_lang: ['AffS1', 'AffS2edit', '1', '3'],
        phone: '+43 123 45678963421',
        salutation: '',
        title_salutation: '',
        title_pre: '',
        title_post: '',
        type: 'u',
        avatar: '',
        website: '',
      },
    },
    description: 'test',
    display: 'Check-text',
    last_modified_at: '2019-06-18T09:57:24.704395+02:00',
    last_modified_by: {
      pk: 150,
      username: 'testuser',
      email: 'testuser@domain.com',
      is_active: true,
      is_staff: true,
      last_login: '2020-07-02T15:06:38.181948+02:00',
      userprofile: {
        first_name: 'Test',
        last_name: 'User',
        anonymized: false,
        academic_title: 'Test',
        additional_information: 'Something about me',
        country: 'Österreich',
        email_others: [],
        org_zug_mitarbeiter: [],
        org_zug_mitarbeiter_lang: ['Aff1edit', 'Aff1', '2', '3'],
        org_zug_student: [],
        org_zug_student_lang: ['AffS1', 'AffS2edit', '1', '3'],
        phone: '+43 123 45678963421',
        salutation: '',
        title_salutation: '',
        title_pre: '',
        title_post: '',
        type: 'u',
        avatar: '',
        website: '',
      },
    },
    name: 'Check-text',
    pk: '9d33359a-afd7-4149-bbe0-95f819d6bebd',
    type_settings: {
      final: true,
      answers: ['Check me'],
      multiple_select: true,
    },
  },
  ordering: 1,
  pk: 'df916528-90cf-4e00-9353-f44123c74711',
  values: {
    answers: [
      {
        answer: 'Check me',
        selected: true,
      },
    ],
  },
};

export const mockMetadataParamData: ParametersData = {
  'e22cf025-c10f-4097-aef7-d9ea9a802562': mockMetadata,
  '9d33359a-afd7-4149-bbe0-95f819d6bebd': mockMetadata2,
};
