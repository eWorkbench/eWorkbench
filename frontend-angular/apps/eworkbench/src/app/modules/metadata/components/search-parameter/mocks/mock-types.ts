/**
 * Copyright (C) 2016-2020 TU Muenchen and contributors of ANEXIA Internetdienstleistungs GmbH
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import type { DropdownElement } from '@eworkbench/types';

export const mockTypes: DropdownElement[] = [
  { value: '27b398b6-c7b4-4746-98db-4e527d8b1919', label: 'Active' },
  { value: '44f035e9-229d-48cd-86c3-8fbe1bcea00c', label: 'Selection' },
  { value: 'e85cd7a0-72b1-4563-9014-f703931cd19f', label: 'Integer' },
];
