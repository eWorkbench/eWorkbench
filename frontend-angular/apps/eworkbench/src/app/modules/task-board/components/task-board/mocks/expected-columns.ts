/* istanbul ignore file */

/**
 * Copyright (C) 2016-2020 TU Muenchen and contributors of ANEXIA Internetdienstleistungs GmbH
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import { TaskBoardColumn } from '@eworkbench/types';

export const expectedColumns: TaskBoardColumn[] = [
  {
    title: 'New column',
    display: 'Kanban Board Column New column belonging to Board Test',
    pk: '24bbca96-da34-4dfd-9ce5-88295152ef1f',
    color: 'rgba(224,224,224,0.65)',
    content_type: 44,
    icon: '',
    content_type_model: 'kanban_boards.kanbanboardcolumn',
    ordering: 1,
    tasks: [
      {
        task_id: '16820249-90ce-48bd-846e-9ecdbaf0251f',
        num_related_notes: 0,
        task: {
          state: 'NEW',
          metadata: [],
          version_number: 0,
          assigned_users: [],
          task_id: 119,
          description: '',
          due_date: null,
          created_at: '2020-05-10T16:34:10.566222+02:00',
          url: '',
          pk: '16820249-90ce-48bd-846e-9ecdbaf0251f',
          content_type: 33,
          deleted: false,
          labels: [],
          priority: 'NORM',
          start_date: null,
          checklist_items: [],
          title: 'Test',
          last_modified_by: {
            pk: 159,
            username: 'e2e',
            email: '',
            is_active: true,
            is_staff: false,
            last_login: '2020-05-26T11:17:47.030000+02:00',
            userprofile: {
              first_name: '',
              last_name: '',
              anonymized: false,
              academic_title: '',
              additional_information: '',
              country: '',
              email_others: [],
              org_zug_mitarbeiter: [],
              org_zug_mitarbeiter_lang: [],
              org_zug_student: [],
              org_zug_student_lang: [],
              phone: '',
              salutation: '',
              title_salutation: '',
              title_pre: '',
              title_post: '',
              type: 'u',
              avatar: '',
              website: null,
            },
          },
          last_modified_at: '2020-05-10T16:34:10.669981+02:00',
          display: 'Test',
          projects: [],
          created_by: {
            pk: 159,
            username: 'e2e',
            email: '',
            is_active: true,
            is_staff: false,
            last_login: '2020-05-26T11:17:47.030000+02:00',
            userprofile: {
              first_name: '',
              last_name: '',
              anonymized: false,
              academic_title: '',
              additional_information: '',
              country: '',
              email_others: [],
              org_zug_mitarbeiter: [],
              org_zug_mitarbeiter_lang: [],
              org_zug_student: [],
              org_zug_student_lang: [],
              phone: '',
              salutation: '',
              title_salutation: '',
              title_pre: '',
              title_post: '',
              type: 'u',
              avatar: '',
              website: null,
            },
          },
          assigned_users_pk: [],
          content_type_model: 'shared_elements.task',
        },
        display: 'Task Test is assigned to column New column',
        url: '',
        pk: 'c9fb3fa6-88c2-4b35-84ef-da512b27ae1e',
        content_type: 45,
        kanban_board_column: '24bbca96-da34-4dfd-9ce5-88295152ef1f',
        content_type_model: 'kanban_boards.kanbanboardcolumntaskassignment',
        ordering: 0,
      },
    ],
  },
  {
    title: 'New column 2',
    display: 'Kanban Board Column New column belonging to Board Test',
    pk: '761163de-fdef-43f1-9324-73094e94fa08',
    color: 'rgba(224,224,224,0.65)',
    content_type: 44,
    icon: '',
    content_type_model: 'kanban_boards.kanbanboardcolumn',
    ordering: 2,
    tasks: [
      {
        task_id: '23d8a0fd-74cb-4fa5-a042-3064a4a13163',
        num_related_notes: 0,
        task: {
          state: 'NEW',
          metadata: [],
          version_number: 0,
          assigned_users: [],
          task_id: 120,
          description: '',
          due_date: null,
          created_at: '2020-05-10T16:34:59.181506+02:00',
          url: '',
          pk: '23d8a0fd-74cb-4fa5-a042-3064a4a13163',
          content_type: 33,
          deleted: false,
          labels: [],
          priority: 'NORM',
          start_date: null,
          checklist_items: [],
          title: 'Test 3',
          last_modified_by: {
            pk: 159,
            username: 'e2e',
            email: '',
            is_active: true,
            is_staff: false,
            last_login: '2020-05-26T11:17:47.030000+02:00',
            userprofile: {
              first_name: '',
              last_name: '',
              anonymized: false,
              academic_title: '',
              additional_information: '',
              country: '',
              email_others: [],
              org_zug_mitarbeiter: [],
              org_zug_mitarbeiter_lang: [],
              org_zug_student: [],
              org_zug_student_lang: [],
              phone: '',
              salutation: '',
              title_salutation: '',
              title_pre: '',
              title_post: '',
              type: 'u',
              avatar: '',
              website: null,
            },
          },
          last_modified_at: '2020-05-10T16:34:59.281549+02:00',
          display: 'Test 3',
          projects: [],
          created_by: {
            pk: 159,
            username: 'e2e',
            email: '',
            is_active: true,
            is_staff: false,
            last_login: '2020-05-26T11:17:47.030000+02:00',
            userprofile: {
              first_name: '',
              last_name: '',
              anonymized: false,
              academic_title: '',
              additional_information: '',
              country: '',
              email_others: [],
              org_zug_mitarbeiter: [],
              org_zug_mitarbeiter_lang: [],
              org_zug_student: [],
              org_zug_student_lang: [],
              phone: '',
              salutation: '',
              title_salutation: '',
              title_pre: '',
              title_post: '',
              type: 'u',
              avatar: '',
              website: null,
            },
          },
          assigned_users_pk: [],
          content_type_model: 'shared_elements.task',
        },
        display: 'Task Test 3 is assigned to column New column',
        url: '',
        pk: 'c80b489a-0fc9-4073-b4c4-89bf6af3205f',
        content_type: 45,
        kanban_board_column: '761163de-fdef-43f1-9324-73094e94fa08',
        content_type_model: 'kanban_boards.kanbanboardcolumntaskassignment',
        ordering: 0,
      },
    ],
  },
];
