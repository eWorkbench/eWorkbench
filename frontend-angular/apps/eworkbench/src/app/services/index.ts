/**
 * Copyright (C) 2016-2020 TU Muenchen and contributors of ANEXIA Internetdienstleistungs GmbH
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

export * from './app-version/app-version.service';
export * from './appointments/appointments.service';
export * from './auth/auth.service';
export * from './calendar-access-privileges/calendar-access-privileges.service';
export * from './contact-form/contact-form.service';
export * from './contacts/contacts.service';
export * from './content-type-model/content-type-model.service';
export * from './dashboard/dashboard.service';
export * from './dmp/dmp.service';
export * from './drives/drives.service';
export * from './dss-containers/dss-containers.service';
export * from './files/files.service';
export * from './labbook-sections/labbook-sections.service';
export * from './labbooks/labbooks.service';
export * from './labels/labels.service';
export * from './menu-model/menu-model.service';
export * from './metadata/metadata.service';
export * from './my-resource-bookings/my-resource-bookings.service';
export * from './my-schedule/my-schedule.service';
export * from './notes/notes.service';
export * from './notification-configuration/notification-configuration.service';
export * from './page-title/page-title.service';
export * from './password/password.service';
export * from './pictures/pictures.service';
export * from './plugin-instances/plugin-instances.service';
export * from './plugins/plugins.service';
export * from './privileges/privileges.service';
export * from './projects-sidebar-model/projects-sidebar-model.service';
export * from './projects/projects.service';
export * from './resource-bookings/resource-bookings.service';
export * from './resources/resources.service';
export * from './roles/roles.service';
export * from './search/search.service';
export * from './site-preferences/site-preferences.service';
export * from './task-board/task-board.service';
export * from './tasks-backlog/tasks-backlog.service';
export * from './tasks/tasks.service';
export * from './user-groups/user-groups.service';
export * from './websocket/websocket.service';
