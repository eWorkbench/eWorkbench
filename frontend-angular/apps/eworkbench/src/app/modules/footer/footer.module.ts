/* istanbul ignore file */

/**
 * Copyright (C) 2016-2020 TU Muenchen and contributors of ANEXIA Internetdienstleistungs GmbH
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { FooterComponent } from './components/footer/footer.component';
import { RouterModule } from '@angular/router';
import { TranslocoRootModule } from '@app/transloco-root.module';

@NgModule({
  declarations: [FooterComponent],
  imports: [CommonModule, RouterModule, TranslocoRootModule],
  exports: [FooterComponent],
})
export class FooterModule {}
