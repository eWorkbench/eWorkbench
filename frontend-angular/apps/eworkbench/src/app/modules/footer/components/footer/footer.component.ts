/**
 * Copyright (C) 2016-2020 TU Muenchen and contributors of ANEXIA Internetdienstleistungs GmbH
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import { ChangeDetectorRef, Component, OnInit } from '@angular/core';
import { AuthService } from '@app/services';
import { User } from '@eworkbench/types';
import { TranslocoService } from '@ngneat/transloco';
import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';

@UntilDestroy()
@Component({
  selector: 'eworkbench-footer',
  templateUrl: './footer.component.html',
  styleUrls: ['./footer.component.scss'],
})
export class FooterComponent implements OnInit {
  public currentUser: User | null = null;

  public constructor(
    private readonly authService: AuthService,
    public readonly translocoService: TranslocoService,
    public readonly cdr: ChangeDetectorRef
  ) {}

  public ngOnInit(): void {
    this.authService.user$.pipe(untilDestroyed(this)).subscribe(state => {
      this.currentUser = state.user;
      this.cdr.markForCheck();
    });
  }
}
