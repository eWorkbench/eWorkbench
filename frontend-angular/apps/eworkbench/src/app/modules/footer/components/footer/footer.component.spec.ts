/**
 * Copyright (C) 2016-2020 TU Muenchen and contributors of ANEXIA Internetdienstleistungs GmbH
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import { AuthService } from '@app/services';
import { getTranslocoModule } from '@app/transloco-testing.module';
import { mockUser } from '@eworkbench/mocks';
import { mockProvider } from '@ngneat/spectator';
import { createRoutingFactory, SpectatorRouting } from '@ngneat/spectator/jest';
import { of } from 'rxjs';
import { FooterComponent } from './footer.component';

describe('FooterComponent', () => {
  let spectator: SpectatorRouting<FooterComponent>;
  const createComponent = createRoutingFactory({
    component: FooterComponent,
    imports: [getTranslocoModule()],
    providers: [
      mockProvider(AuthService, {
        user$: of(mockUser),
        login: () => of(mockUser),
      }),
    ],
  });

  beforeEach(() => (spectator = createComponent()));

  it('should create', () => {
    expect(spectator).toBeTruthy();
  });

  it('should have the proper brand name', () => {
    const span = spectator.query<HTMLSpanElement>('.navbar-brand');
    expect(span).toHaveText('Universitätsbibliothek der Technischen Universität München');
  });

  it('should match snapshot', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });
});
