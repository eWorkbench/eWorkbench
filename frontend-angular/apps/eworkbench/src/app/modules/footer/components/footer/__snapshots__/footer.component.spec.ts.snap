// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterComponent should match snapshot 1`] = `
<eworkbench-footer
  __ngContext__={[Function LRootView]}
  authService={[Function Object]}
  cdr={[Function ViewRef]}
  currentUser="undefined"
  translocoService={[Function TranslocoService]}
>
  <footer
    class="navbar navbar-expand-lg py-0"
  >
    <span
      class="navbar-brand"
    >
      Universitätsbibliothek der Technischen Universität München
    </span>
    <ul
      class="navbar-nav ml-auto h-100"
    >
      
      <li
        class="h-100 d-flex align-items-center"
      >
        <a
          class="h-100 w-100"
        >
          <button
            class="btn ewb-button-regular font-weight-normal h-100 w-100 text-left"
          >
             FAQ 
          </button>
        </a>
      </li>
      <li
        class="h-100 d-flex align-items-center"
      >
        <a
          class="h-100 w-100"
        >
          <button
            class="btn ewb-button-regular font-weight-normal h-100 w-100 text-left"
          >
             Privacy Policy 
          </button>
        </a>
      </li>
      <li
        class="h-100 d-flex align-items-center"
      >
        <a
          class="h-100 w-100"
        >
          <button
            class="btn ewb-button-regular font-weight-normal h-100 w-100 text-left"
          >
             Imprint 
          </button>
        </a>
      </li>
      <li
        class="h-100 d-flex align-items-center"
      >
        <a
          class="h-100 w-100"
        >
          <button
            class="btn ewb-button-regular font-weight-normal h-100 w-100 text-left"
          >
             OSS Licenses 
          </button>
        </a>
      </li>
      <li
        class="h-100 d-flex align-items-center"
      >
        <a
          class="h-100 w-100"
          href="https://github.com/eworkbench/"
          target="_blank"
        >
          <button
            class="btn ewb-button-regular font-weight-normal h-100 w-100 text-left"
          >
             GitHub 
          </button>
        </a>
      </li>
    </ul>
  </footer>
</eworkbench-footer>
`;
